app/routers/match_router.py

```diff
@@ -49,17 +69,12 @@ async def get_match(data: MatchQuery):
     """
     if data.user_type == "mentor":
         collection = "Mentees"
-        user_query = {"profile_id": {"$in": Router.db.first(
-            "Matches", {"mentor_id": data.user_id})['mentee_ids']}}
     elif data.user_type == "mentee":
         collection = "Mentors"
-        user_query = {
-            "profile_id": {
-                "$in": [mentor["mentor_id"] for mentor in Router.db.read(
-                    "Matches", {"mentee_ids": data.user_id})]
-            }
-        }
     else:
         raise ValueError("get_match: user_type must be 'mentor' or 'mentee'")
 
+    match_ids = get_match_ids(data.user_id, data.user_type)
+    user_query = {"profile_id": {"$in": match_ids}}
```

Comment:
I'm definitely taking notes from the way you work. I want my work to be that good. Love it!